File: codegen/results/v002/ollama/qwen3.5_27b-nvfp4_nothink/20260412_033843/go/005_unit_test_writer/solution.go
package main

import "fmt"

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func runTest(description string, price float64, quantity int, expected float64) {
	result := calculateDiscount(price, quantity)
	if result == expected {
		fmt.Printf("PASS: %s\n", description)
	} else {
		fmt.Printf("FAIL: %s (expected: %.2f, got: %.2f)\n", description, expected, result)
	}
}

func main() {
	// Test Case 1: Boundary - quantity < 10 (no discount)
	runTest("Quantity 9 (no discount)", 10.0, 9, 90.0)

	// Test Case 2: Boundary - quantity >= 10 and < 50 (10% discount)
	runTest("Quantity 10 (10%% discount)", 10.0, 10, 90.0)

	// Test Case 3: Boundary - quantity just below 50 (10% discount)
	runTest("Quantity 49 (10%% discount)", 10.0, 49, 441.0) // 10 * 49 * 0.9 = 441

	// Test Case 4: Boundary - quantity >= 50 (25% discount)
	runTest("Quantity 50 (25%% discount)", 10.0, 50, 375.0) // 10 * 50 * 0.75 = 375

	// Additional Test Case: Non-boundary value in the middle range
	runTest("Quantity 25 (10%% discount)", 20.0, 25, 450.0) // 20 * 25 * 0.9 = 450
}

File: codegen/results/v002/ollama/qwen3.5_27b-nvfp4_nothink/20260412_033843/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    {
        description: "Quantity less than 10 (no discount)",
        price: 10,
        quantity: 9,
        expected: 90 // 10 * 9
    },
    {
        description: "Boundary at quantity 10 (10% discount)",
        price: 10,
        quantity: 10,
        expected: 90 // 10 * 10 * 0.9
    },
    {
        description: "Quantity less than 50 but >= 10 (10% discount)",
        price: 20,
        quantity: 49,
        expected: 882 // 20 * 49 * 0.9
    },
    {
        description: "Boundary at quantity 50 (25% discount)",
        price: 20,
        quantity: 50,
        expected: 750 // 20 * 50 * 0.75
    }
];

function runTests() {
    let passedCount = 0;
    let failedCount = 0;

    for (const testCase of testCases) {
        const result = calculateDiscount(testCase.price, testCase.quantity);
        
        // Use a small epsilon for floating point comparison
        const epsilon = 0.000001;
        const isPassing = Math.abs(result - testCase.expected) < epsilon;

        if (isPassing) {
            console.log(`PASS: ${testCase.description}`);
            passedCount++;
        } else {
            console.log(`FAIL: ${testCase.description} (expected: ${testCase.expected}, got: ${result})`);
            failedCount++;
        }
    }

    // Print a summary
    console.log(`\nTotal: ${passedCount + failedCount} | Passed: ${passedCount} | Failed: ${failedCount}`);
    
    // Exit with non-zero code if any tests failed
    if (failedCount > 0) {
        process.exit(1);
    }
}

runTests();